### ISSUE TITLE
breakout installation in favor of individual guides

### ISSUE THREAD

**Issue by alisondy** [fluxcd/website#523]
this is a tracking issue

Current PRS in Flight

#521 Install Guides
#520 Uninstall Guide
#519 Upgrade Guide
#518 Bootstrap guides

TODO
- [ ] Redo page weights
- [ ] Remove installation.md - Blocked by #521,#520,#519,#518 

notes
- Redirects for installation.md may end up being alot of work, maybe investigate creating a 404 page with links to common areas

**Comment by stefanprodan** [comment]
I propose we make installation into a section with cli/bootstrap/upgrade/uninstall as subsections. This way no links will break and we don't hide the install content under guides.

**Comment by alisondy** [comment]
> I propose we make installation into a section with cli/bootstrap/upgrade/uninstall as subsections. This way no links will break and we don't hide the install content under guides.

- installing the CLI is  a prereq to running tasks like bootstrap, upgrade, and uninstall.
- tasks like bootstrap, upgrade, and uninstall pertain to the lifecycle of a flux deployment

Perhaps something like?

 /install-the-flux-cli
 /Deploy and Manage Flux Components
      /bootstrap flux
      /upgrade flux
      /uninstall flux
      /customize flux

**Comment by stefanprodan** [comment]
> installing the CLI is a prereq to running tasks like bootstrap, upgrade, and uninstall.

Yes we can write the CLI steps in the `installation/index.md` then all others will be under installation.

**Comment by alisondy** [comment]
ok sounds good

**Comment by alisondy** [comment]
no doesn't sound good actually
- Windows OSX and Linux are separate guides
this is because in the former format people had to sift through all the different options to find the ones relevant to their operating system

if we collate them all into one page it'll be long and repetitive.

**Comment by stefanprodan** [comment]
The last structure you posted looks good to me 👍

**Comment by stefanprodan** [comment]
Another option would be:

```
/installation
  _index.md
  /cli
    macos.md
    linux.md
    windows.md
 /bootstrap
 /upgrade
 /uninstall
```

**Comment by alisondy** [comment]
When you mentioned previous structure, were you referring to this one?
I'd be happy to ahead with this:

```
 /Install-the-flux-cli
 /Deploy and Manage Flux
      /bootstrap flux
      /upgrade flux
      /uninstall flux
      /customize flux
```

**Comment by scottrigby** [comment]
This also lgtm, with just a small change.

Remember when we were looking at the AWS guides model, where the pages are shorter and so when you search you see a list of results specific to what you need? I feel the structure above is very similar to that strategy and if we roll that out across the docs that could be a very good thing for end users.

The `/Deploy and Manage Flux Components` section lgtm from this angle, but suggesting one small adjustment to pages under `/Install-the-flux-cli`:

```diff
 /Install-the-flux-cli
       _index.md
-       windows
-       osx
-       linux
+       Install CLI on windows
+       Install CLI on osx
+       Install CLI on linux
 /Deploy and Manage Flux Components
```

This way when someone types `install` or `cli` into search, they'll see a dropdown select of matching pages like:
```
"CLI" or "install"
    install CLI on windows
    install CLI on osx
    install CLI on linux
```
instead of only:
```
"CLI" or "install"
    windows
    osx
    linux
```

**Comment by alisondy** [comment]
Update on this work
- need to reweight all the pages
can a seprate branch be made to take in all 4 of these major changes,
then a pr onto that branch to redo page weights
then a pr from that branch into main to apply new page weights and pages.

**Comment by alisondy** [comment]
Changed the approach

Made every section of major changes into individual commits
- install
- upgrade
- bootstrap
- uninstall
- remove installation.md
- reweight

Can review each commit individually instead of having to review 4 different PR's

https://github.com/fluxcd/website/pull/530

If you don't want to be overwhelmed by every single file edited, I recommend browsing by individual commit for each of the changes.

**Comment by kingdonb** [comment]
The installation guide has been updated to match the idea proposed here,

https://fluxcd.io/flux/installation/

Instead of telling everything, it covers just the prerequisites, the CLI installation, and an overview of bootstrap with links to the specific guides. All of the details that are cloud-provider specific or repository-host specific are now contained within the sub-guides, which are each comprehensive.

(I'm going through the notes for GSoD and issues linked in https://fluxcd.io/contributing/docs/google-season-of-docs-2023/ to figure out what parts remain that we still need to focus on.)
